docs/application/web/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,445 @@
+   ```javascript
+   var inputInfo = src.inputTensorsInfo;
+
+   var inputData = inputInfo.getTensorsData();
+   var rawInputData = inputData.getTensorRawData(0);
+   inputData.setTensorRawData(0, [123]);
+
+   src.inputData(inputData);
+   ```
+
+Input data will be passed to the further stages of the pipeline.
+
+
+## Read data from pipeline output
+You can read the tensors output by the pipeline with the application code by registering a callback, triggered by data coming into `appsink` or `tensor_sink`.
+Use `gst-inspect-1.0` to learn more about the differences between these to elements.
+
+To get pipeline output:
+1. Create a pipeline with a sink node that can pass data to application:
+   ```javascript
+   var pipelineDescription = 'videotestsrc num-buffers=3 ' +
+                             '! videoconvert ' +
+                             '! videoscale ' +
+                             '! video/x-raw,format=RGBx,width=16,height=16,framerate=10/1 ' +
+                             '! tensor_converter ' +
+                             '! tensor_sink name=sinkx';
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
+
+2. Register a listener triggered by data incoming to `sinkx`:
+   ```javascript
+   function sinkListener(sinkName, data) {
+      console.log('SinkListener for ' + sinkName + ' sink called. Data dimensions: ' +
+              data.tensorsInfo.getDimensions(0));
+
+      // Read and process data according to your needs
+   }
+   pipeline.registerSinkListener('sinkx', sinkListener);
+   ```
+
+After starting the pipeline, `sinkListener` will be called repeatedly.
+
+
+## Change data flow within pipeline
+You may want to set the the source of the data to one of several pipeline branches.
+It can be done with `input-selector` node and `Switch` API.
+
+We will use the pipeline from the figure below to show how to use `input-selector`, to choose between blue and green branch as the source for the white branch.
+
+![Pipeline with branches and a switch](./media/pipeline_with_switch.svg)
+
+To choose a source branch with `input-selector`:
+1. Create a pipeline:
+   ```javascript
+   var pipelineDescription = 'input-selector name=ins ! tensor_converter ! tensor_sink name=sinkx ' + // white branch
+                             'videotestsrc is-live=true ! videoconvert ! ins.sink_0 ' + // blue branch
+                             'videotestsrc num-buffers=3 is-live=true ! videoconvert ! ins.sink_1'; // green branch
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
+
+2. Get `insSwitch`:
+   ```javascript
+   var insSwitch = pipeline.getSwitch('ins');
+   ```
+
+3. You can get `Switch` pads:
+   ```javascript
+   console.log(insSwitch.getPadList()); // ["sink_0", "sink_1"]
+   ```
+
+4. Choose the pad to be used as a source for `input-selector` and the white pipeline branch:
+   ```javascript
+   insSwitch.select('sink_1'); // green branch used as the source
+   ```
```

Comment:
```suggestion
## Change data flow within pipeline

You may want to set the the source of the data to one of several pipeline branches.
It can be done with `input-selector` node and `Switch` API. Use the pipeline from the following figure to show how to use `input-selector` to choose between blue and green branch as the source for the white branch.

![Pipeline with branches and a switch](./media/pipeline_with_switch.svg)

To choose a source branch with `input-selector`:

1. Create a pipeline:
  
    ```javascript
   var pipelineDescription = 'input-selector name=ins ! tensor_converter ! tensor_sink name=sinkx ' + // white branch
                             'videotestsrc is-live=true ! videoconvert ! ins.sink_0 ' + // blue branch
                             'videotestsrc num-buffers=3 is-live=true ! videoconvert ! ins.sink_1'; // green branch
   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
   ```
2. Get `insSwitch`:
  
   ```javascript
   var insSwitch = pipeline.getSwitch('ins');
   ```

3. Get `Switch` pads:
   
    ```javascript
       console.log(insSwitch.getPadList()); // ["sink_0", "sink_1"]
   ```
4. Choose the pad to be used as a source for `input-selector` and the white pipeline branch:
   
      ```javascript
      insSwitch.select('sink_1'); // green branch used as the source
      ```